rancher2/resource_rancher2_cluster.go

```diff
@@ -398,3 +402,12 @@ func createClusterRegistrationToken(client *managementClient.Client, clusterID s
 	}
 	return newRegToken, nil
 }
+
+func windowsPreferredClusterCustomDiffHandler()  schema.CustomizeDiffFunc{
+	//Cluster must be recreated to implement this if there is a change to windows var if the cluster is not
+	//already a windows cluster. Currently this is unsupported by rancher, so force new resource here.
+	//https://rancher.com/docs/rancher/v2.x/en/cluster-provisioning/rke-clusters/windows-clusters/
+	return customdiff.ForceNewIfChange("windows_prefered_cluster", func (old, new, meta interface{}) bool {
+		return new.(bool) != old.(bool)
+	})
+}
```

Comment:
If a new object needs to be generated on this field change, i think is just enough to add `Forcenew: true` on the field definition. Not needed a cutomizediff function.

Reply:
Gotcha, thanks!